R/system.R

```diff
@@ -1017,9 +1076,12 @@ queryODBC <- function(dsn,username, password, additionalParams, numOfRows = 0, q
   }, error = function(err) {
     # for some cases like conn not being an open connection, sqlQuery still throws error. handle it here.
     # clear connection in pool so that new connection will be used for the next try
-    clearDBConnection("odbc", NULL, NULL, NULL, username, dsn = dsn, additionalParams = additionalParams)
+    clearDBConnection(type, NULL, NULL, NULL, username, dsn = dsn, additionalParams = additionalParams)
     stop(err)
   })
+  if(type == "mssqlserver") {
+    DBI::dbClearResult(resultSet)
```

Comment:
Please explain why this is necessary only for mssqlserver case, if that is the case.

Reply:
Added comment.